src/benchmarklib/benchmark_runner.cpp

```diff
@@ -438,7 +444,9 @@ void BenchmarkRunner::_warmup(const BenchmarkItemID item_id) {
 
   // Wait for the rest of the tasks that didn't make it in time.
   Hyrise::get().scheduler()->wait_for_all_tasks();
+
   Assert(_currently_running_clients == 0, "All runs must be finished at this point.");
+  deplete_semaphore(_running_clients_semaphore);
```

Comment:
Do we have to wait here? From my intuition, there should not be any clients running at this point.

Reply:
Also, though it's not 100% the same, we do not really need `_currently_running_clients` and could just change such calls to `Assert(_running_clients_semaphore.try_acquire(), ...);` (plus release). Not sure if that's something we want.

Reply:
(discussed offline)